src/gui/editors/BBEditor.cpp

```diff
@@ -264,3 +261,27 @@ void BBTrackContainerView::updatePosition()
 	//realignTracks();
 	emit positionChanged( m_currentPosition );
 }
+
+
+
+
+void BBTrackContainerView::makeSteps( bool clone )
+{
+	TrackContainer::TrackList tl = model()->tracks();
+
+	for( TrackContainer::TrackList::iterator it = tl.begin();
+		it != tl.end(); ++it )
+	{
+		if( ( *it )->type() == Track::InstrumentTrack )
+		{
+			Pattern* p = static_cast<Pattern *>( ( *it )->getTCO( m_bbtc->currentBB() ) );
+			if( clone )
```

Comment:
Well, on the other hand one could argue that you would remove one method and layer of indirection.

I've seen this pattern repeatedly on LMMS, and find it particularly confusing because it's not always obvious what the parameter does. Anyway, it's fine for me if this gets in though, not something that I strongly disagree.